android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.kt

```diff
@@ -185,21 +185,42 @@ class MusicModule(private val reactContext: ReactApplicationContext?) :
     fun add(data: ReadableArray?, insertBeforeIndex: Int, callback: Promise) {
         if (verifyServiceBoundOrReject(callback)) return
 
+        val tracks: MutableList<Track> = mutableListOf()
         val bundleList = Arguments.toList(data)
-        val tracks: List<Track> = try {
-            Track.createTracks(
-                reactApplicationContext, bundleList, RatingCompat.RATING_HEART
-            )!!
-
-
-        } catch (ex: Exception) {
-            callback.reject("invalid_track_object", ex)
-            return
+        if (bundleList != null) {
+            for (trackBundle in bundleList) {
+                if (trackBundle is Bundle) {
+                    try {
+                        tracks.add(
+                            Track(reactApplicationContext, trackBundle, RatingCompat.RATING_HEART)
+                        )
+                    } catch(ex: Exception) {
+                        callback.reject("invalid_track_object", ex)
+                    }
+                } else {
+                    callback.reject("invalid_track_object", "Track was not a dictionary type")
+                }
+            }
+        } else {
+            callback.reject("invalid_parameter", "Was not given an array of tracks")
```

Comment:
It looks like there are quite a few nested calls here. Maybe it can be simplified.

Reply:
Do you have a suggestion?

Reply:
You can do something like this:

```kotlin
        if (bundleList == null) {
            callback.reject("invalid_parameter", "Was not given an array of tracks")
            return
        }

        bundleList.forEach {
            if (it is Bundle) {
                tracks.add(Track(reactApplicationContext, it, RatingCompat.RATING_HEART))
            } else {
                callback.reject("invalid_track_object", "Track was not a dictionary type")
            }
        }
```

You can flip the null check and also remove the `try catch` as I don't think anything can throw an exception inside `Track`

Reply:
cool. I like this! I had the try catch cause it was there before and wasn't sure if something could throw.